### PR TITLE
Add web-dev and game-dev levels

### PR DESCRIPTION
**Pull request by nwinter**
Almost all work is done on web-dev v1. game-dev v1 is further from done but useful to have live for development. The web-dev and game-dev courses don't load for non-admins (that part is already in master), so it's not really user-facing yet.

I also ripped out a bunch of old code while I was it it, including real-time multiplayer prototype, multiple spells' interface code, and Firebase.

I know code review on this whole thing would be a lot for one sitting, but depending on what you worked on, you might have interest in checking out:

* The changes to TeacherClassView
* The way we now use `level.isType()` instead of comparing `level.get('type')`
* How WebSurfaceView, web-dev-iframe.html, and web-dev-listener.js work
* Changes to how the CSV exports, TeacherClassView, and TeacherClassesView acronymize courses (CS1, CS2, GD1, WD1, CS3, etc.)
* Courses are now returned from the server and added to classrooms in an order specified in the Course server model

Preview: ;)
screenshot 2016-07-17 01 23 34

### CODE REVIEW

File: server/models/Course.coffee [hunk 1]
```diff
@@ -13,4 +13,32 @@ CourseSchema.statics.editableProperties = []
...
 CourseSchema.statics.jsonSchema = jsonSchema
 
+CourseSchema.statics.sortCourses = (courses) ->
+  ordering = [
+    'introduction-to-computer-science'
+    'computer-science-2'
+    'game-dev-1'
+    'web-dev-1'
+    'computer-science-3'
+    'game-dev-1'
```
Comment (Imperadeiro98) [inline server/models/Course.coffee]: Shouldn't this line be game-dev-2?

Comment (nwinter) [inline server/models/Course.coffee]: Yes!

File: app/schemas/models/level_session.coffee [hunk 2]
```diff
@@ -37,8 +37,6 @@ _.extend LevelSessionSchema.properties,
     type: 'string'
   levelID:
     type: 'string'
-  multiplayer:
```
Comment (differentmatt) [inline app/schemas/models/level_session.coffee]: We'll need to unset multiplayer on existing level sessions so they still pass validation after this change.

Comment (nwinter) [inline app/schemas/models/level_session.coffee]: I already unset almost all of them. There may be some code live that continues to set it, so we could pre-deploy the removal of that, remove the flags, and then deploy live, or we could just deploy live and at the same time run this script (shouldn't take long now):

```
db.level.sessions.update({multiplayer: {$exists: true}}, {$unset: {multiplayer: 1}}, {multi: true})
```

Comment (differentmatt) [inline app/schemas/models/level_session.coffee]: Ah, excellent. Let's run the script again after deploy.

**Pull request merged**